### PR TITLE
common/options: document some osd/mon options

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/common/options.cc [hunk 1]
```diff
@@ -2247,77 +2248,82 @@ std::vector<Option> get_global_options() {
...
     Option("osd_max_backfills", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
     .set_default(1)
-    .set_description(""),
+    .set_description("maximum number of concurrent backfills per OSD"),
 
     Option("osd_min_recovery_priority", Option::TYPE_INT, Option::LEVEL_ADVANCED)
     .set_default(0)
-    .set_description(""),
+    .set_description("minimum priority below which recovery is not performed")
+    .set_long_description("The purpose here is to prevent the cluster from performany *any* low priority work (e.g., rebalancing) below this threshold and focus solely on higher priority work (e.g., replicated degraded objects)."),
```
Comment (tchaikov) [inline src/common/options.cc]: s/performany/perform/

Comment (tchaikov) [reply src/common/options.cc]: "replicated degraded objects" does not sound right, how about "replicating degraded objects"?

File: src/common/options.cc [hunk 2]
```diff
@@ -2247,77 +2248,82 @@ std::vector<Option> get_global_options() {
...
     Option("osd_max_backfills", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
     .set_default(1)
-    .set_description(""),
+    .set_description("maximum number of concurrent backfills per OSD"),
 
     Option("osd_min_recovery_priority", Option::TYPE_INT, Option::LEVEL_ADVANCED)
     .set_default(0)
-    .set_description(""),
+    .set_description("minimum priority below which recovery is not performed")
+    .set_long_description("The purpose here is to prevent the cluster from perform *any* low priority work (e.g., rebalancing) below this threshold and focus solely on higher priority work (e.g., replicating degraded objects)."),
```
Comment (neha-ojha) [inline src/common/options.cc]: s/perform/performing/

File: src/common/options.cc [hunk 3]
```diff
@@ -2247,77 +2248,82 @@ std::vector<Option> get_global_options() {
...
     .set_default(5.0)
-    .set_description(""),
+    .set_description("how long agent should sleep if it has no work to do"),
 
-    Option("osd_find_best_info_ignore_history_les", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
+    Option("osd_find_best_info_ignore_history_les", Option::TYPE_BOOL, Option::LEVEL_DEV)
     .set_default(false)
-    .set_description(""),
+    .set_description("ignore last_epoch_started value when peering AND PROBABLY LOSE DATA")
+    .set_long_description("THIS IS AN EXTREMELY DANGEROUS OPTION THAT SHOULD ONLY BE USED AT THE DIRECTION OF A DEVELOPER.  It makes peering ignore the last_epocch_started value when peering, which can allow the OSD to believe an OSD has an authoritative view of a PG's contents even when it is in fact old and stale, typically leading to data loss (by believing a stale PG is up to date)."),
```
Comment (neha-ojha) [inline src/common/options.cc]: s/last_epocch_started/last_epoch_started

**Pull request merged**